backfill.sql

```diff
@@ -79,17 +79,32 @@ BEGIN
 END;
 $$ LANGUAGE PLPGSQL;
 
-CREATE OR REPLACE FUNCTION move_compression_job(IN hypertable_id int, IN new_time timestamptz, OUT old_time timestamptz) 
+CREATE OR REPLACE FUNCTION move_compression_job(IN hypertable_id int, IN hypertable name, IN new_time timestamptz, OUT old_time timestamptz) 
 AS $$
 DECLARE
     compression_job_id int;
+    version int;
 BEGIN
-    SELECT job_id INTO compression_job_id FROM _timescaledb_config.bgw_policy_compress_chunks b WHERE b.hypertable_id = move_compression_job.hypertable_id;
+    SELECT split_part(extversion, '.', 1)::INT INTO version FROM pg_catalog.pg_extension WHERE extname='timescaledb' LIMIT 1;
+
+    IF version = 1 THEN
+        SELECT job_id INTO compression_job_id FROM _timescaledb_config.bgw_policy_compress_chunks b WHERE b.hypertable_id = move_compression_job.hypertable_id; 
+    ELSE
+        SELECT s.job_id INTO compression_job_id FROM timescaledb_information.jobs j
+          INNER JOIN timescaledb_information.job_stats s ON j.job_id = s.job_id
+          WHERE j.proc_name = 'policy_compression' AND s.hypertable_name = hypertable;
```

Comment:
Hmmm...just noticed that we're only looking at the hypertable name here, I think if we're gonna do this we're gonna need to include the schema name as well so that we can make sure we don't mix up a hypertable in a different schema with the same name...

Reply:
This is great learning experience for me :D I'll fix that too.

Reply:
I added the schema there now, too, and changed the parameter names to be more aligned with what they are called when calling the function. Test in 2.0.0 ok.

One issue when replacing a previous version of the script is that CREATE OR REPLACE will fail when the parameters are different.